src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -231,6 +269,33 @@ public async Task<string> DockerInspect(IExecutionContext context, string docker
             return (await ExecuteDockerCommandAsync(context, "inspect", $"{options} {dockerObject}")).FirstOrDefault();
         }
 
+        public async Task<List<PortMapping>> DockerPort(IExecutionContext context, string containerId)
+        {
+            const string targetPort = "targetPort";
+            const string proto = "proto";
+            const string host = "host";
+            const string hostPort = "hostPort";
+
+            //"TARGET_PORT/PROTO -> HOST:HOST_PORT"
+            string pattern = $"^(?<{targetPort}>\\d+)/(?<{proto}>\\w+) -> (?<{host}>.+):(?<{hostPort}>\\d+)$";
```

Comment:
usually, i am not a fan of Regex at all for simple parsing. :)
make sure you add L0 test around the parsing.

Reply:
I just followed the pattern I saw for checking the Docker version :) 

I factored this out into a util unit so it that will be testable. This also paves the way for refactoring some other code (like checking Docker version) that may have a lot of complicated logic or regex with no way to just inject the data without making a real call to `docker`